docs/concepts/simulation.md

```diff
@@ -0,0 +1,72 @@
+# Cosmos Blockchain Simulator
+
+The Cosmos SDK offers a full fledged simulation framework to fuzz test every message defined by a module.
+
+This functionality is provided by the[`SimApp`](https://github.com/cosmos/cosmos-sdk/blob/master/simapp/app.go),
+which is an dummy application that is used for running the [`simulation`](https://github.com/cosmos/cosmos-sdk/tree/master/x/simulation) module.
+This module defines all the simulation logic as well as the operations for randomized parameters like accounts, balances etc.
+
+## Goals
+
+The blockchain simulator tests how the blockchain application would behave under real life circumstances by generating and sending randomized messages.
+The goal of this is to detect and debug failures that could halt a live chain, by providing logs and statistics about the operations run by the simulator as well as exporting the latest application state when a failure was found.
+
+Its main difference with integration testing is that the simulator app allows you to pass parameters to customize the chain that's being simulated.
+This comes in handy when trying to reproduce bugs that were generated in the provided operations (randomized or not).
+
```

Comment:
Would be cool to have a "How it Works" section, that explains how the simulation works in the back

Reply:
Ok, so like an "Under the hood" kinda thing? This doc is mostly a high-level introduction to the simulator. 

I think the "How it Works" should ideally point to the `x/simulation` specification (which doesn't exist atm).

Reply:
I would ideally like to see a how it works here, but don't think the addition of that should block merge here.

Reply:
Agree. I think we should write this after we conclude the simulation generalization into the module pattern